examples/nextjs-app-router/app/Signup.tsx

```diff
@@ -1,42 +1,59 @@
 "use client";
 
-import { Form, Input } from "../components/Form";
-import { type FieldErrors } from "simple-stack-form/module";
+import { Form, Input, useFormContext, useFormState } from "../components/Form";
 import { signup } from "./page";
 import { signupSubmit } from "./action";
 
-export default function Signup({
-	serverErrors,
-}: { serverErrors?: FieldErrors<typeof signup> }) {
-	// TODO: call React's `useFormState()` to render server errors
-	// Maybe `useFormState()` should be called in `Form.tsx`?
-	// Should we also update `submitStatus` on the context when the form is submitted?
+export default function Signup() {
+	const [state, formAction] = useFormState(signupSubmit);
+
 	return (
-		<Form
-			action={signupSubmit}
-			className="flex flex-col gap-2 items-start"
-			fieldErrors={serverErrors}
-			validator={signup.validator}
-			name="signupReact"
-		>
-			<FormGroup>
-				<label htmlFor="name">Name</label>
-				<Input id="name" {...signup.inputProps.username} />
-			</FormGroup>
-			<FormGroup>
-				<label htmlFor="optIn">Opt in</label>
-				<Input id="optIn" {...signup.inputProps.optIn} />
-			</FormGroup>
-			<button
-				type="submit"
-				className="bg-purple-700 rounded px-5 py-2 disabled:bg-purple-900"
+		<>
+			{state?.data && (
+				<div
+					className="bg-green-100 mb-4 border border-green-400 text-green-700 px-4 py-3 rounded relative"
+					role="alert"
+				>
+					<strong className="font-bold">Success!</strong>
+					<span className="block sm:inline">
+						You have successfully submitted the form.
+					</span>
+				</div>
+			)}
+			<Form
+				action={formAction}
+				className="flex flex-col gap-2 items-start"
+				fieldErrors={state?.fieldErrors}
+				validator={signup.validator}
+				submitted={!!state?.data}
```

Comment:
```suggestion
```

I'd remove the `submitted` state for now. I think we can decouple simple:form's `submitStatus` and use React's `useFormStatus()` instead. For now, we'll just leave the submit status in a half-broken state. It's not required!